sys/net/network_layer/sixlowpan/lowpan.c

```diff
@@ -1107,25 +1104,24 @@ uint8_t lowpan_iphc_encoding(int if_id, const uint8_t *dest, int dest_len,
                  * and possibly the link-layer addresses.*/
                 lowpan_iphc[1] |= 0x03;
             }
-            else if (dest_len == 2 &&
-                     ipv6_buf->destaddr.uint32[2] == HTONL(0x000000ff) &&
-                     ipv6_buf->destaddr.uint16[6] == HTONS(0xfe00) &&
-                     ipv6_buf->destaddr.uint16[7] == *((uint16_t *) dest)) {
-                /* 0 bits. The address is derived using context information
-                 * and possibly the link-layer addresses.*/
-                lowpan_iphc[1] |= 0x03;
-            }
-            else if ((ipv6_buf->destaddr.uint16[4] == 0) &&
-                     (ipv6_buf->destaddr.uint16[5] == 0) &&
-                     (ipv6_buf->destaddr.uint16[6] == 0) &&
-                     ((ipv6_buf->destaddr.uint8[14]) & 0x80) == 0) {
-                /* 49-bit of interface identifier are 0, so we can compress
-                 * source address-iid to 16-bit */
```

Comment:
I'm not sure either. This code was written by @stze when 6LoWPAN HC was still in draft stage (iirc even the 6LoWPAN RFC wasn't out yet... So this might not have been there that time.